### ISSUE TITLE
E   ModuleNotFoundError: No module named during Test

### ISSUE THREAD

**Issue by am1ru1** [sedelmeyer/cc-pyscript#12]
@sedelmeyer , I am getting the following error when trying executing the pytest even with tox. there shouldn't big difference between my py3.8 and the default py3.7, not sure if you have similar experience?

```
(projectkool) /tmp/projectkool $ python -m pytest
============================= test session starts ==============================
platform linux -- Python 3.8.2, pytest-6.2.2, py-1.10.0, pluggy-0.13.1 -- /usr/src/node-red/.local/share/virtualenvs/projectkool-BjpCI20A/bin/python
cachedir: .pytest_cache
rootdir: /tmp/projectkool, configfile: pytest.ini, testpaths: tests/
plugins: cov-2.11.1
collected 0 items / 1 error

==================================== ERRORS ====================================
__________________ ERROR collecting tests/test_projectkool.py __________________
ImportError while importing test module '/tmp/projectkool/tests/test_projectkool.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
/usr/lib/python3.8/importlib/__init__.py:127: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
tests/test_projectkool.py:5: in <module>
    import projectkool.projectkool as projectkool
E   ModuleNotFoundError: No module named 'projectkool'
=========================== short test summary info ============================
ERROR tests/test_projectkool.py
!!!!!!!!!!!!!!!!!!!! Interrupted: 1 error during collection !!!!!!!!!!!!!!!!!!!!
=============================== 1 error in 0.19s ===============================
(projectkool) /tmp/projectkool $
```

**Comment by sedelmeyer** [comment]
Hello @am1ru1 

I was able to reproduce the error you got above when I did not git initializing my project directory.

Do you know if you ran `git init` in your project directory prior to running `pipenv install --dev`?

This requirement is mentioned here: https://sedelmeyer.github.io/cc-pyscript/readme.html#initiate-git-version-control 

...with additional info here: https://sedelmeyer.github.io/cc-pyscript/tutorial.html#versioning-your-project-with-git-tags-and-setuptools-scm 

Because the generated template uses `setuptools-scm` for versioning, it will require the project to be an initialized git repo, or else the module under `src` will never import correctly.

Just note, something I ran into while testing this was that `pipenv` threw some pretty loud exceptions while installing the local editable package (I am guessing it was because my initialized `git` repo had no initial commits in it), but exception/error messages aside, `pipenv` still managed to install the local editable package successfully for me and after I activated my newly created environment with `pipenv shell`, the `pytest` command ran with all passing tests.

Let me know if running `git init` prior to `pipenv install --dev` works for you!

**Comment by am1ru1** [comment]
That's interesting, the python test is running fine after running `git init` first and then `pipenv install --dev`. However secondary issue pops up "ERROR:setuptools_scm.file_finder_git:listing git files failed - pretending there aren't any".

Note I have changed the Pipfile and Tox.ini to py3.8.

```
/tmp/testrepo2 $ git init
Initialized empty Git repository in /tmp/testrepo2/.git/
/tmp/testrepo2 $
/tmp/testrepo2 $
/tmp/testrepo2 $ vi Pipfile
/tmp/testrepo2 $ ls
CHANGELOG.rst  Pipfile        docs           setup.py       tests
LICENSE        README.rst     setup.cfg      src            tox.ini
/tmp/testrepo2 $ vi tox.ini
/tmp/testrepo2 $ pipenv install --dev
Warning: the environment variable LANG is not set!
We recommend setting this in ~/.profile (or equivalent) for proper expected behavior.
Creating a virtualenv for this project...
Pipfile: /tmp/testrepo2/Pipfile
Using /usr/bin/python3.8 (3.8.2) to create virtualenv...
⠏ Creating virtual environment...created virtual environment CPython3.8.2.final.0-64 in 519ms
  creator CPython3Posix(dest=/usr/src/node-red/.local/share/virtualenvs/testrepo2-Ux6A5Ksm, clear=False, no_vcs_ignore=False, global=False)
  seeder FromAppData(download=False, pip=bundle, setuptools=bundle, wheel=bundle, via=copy, app_data_dir=/usr/src/node-red/.local/share/virtualenv)
    added seed packages: pip==21.0.1, setuptools==52.0.0, wheel==0.36.2
  activators BashActivator,CShellActivator,FishActivator,PowerShellActivator,PythonActivator,XonshActivator

✔ Successfully created virtual environment!
Virtualenv location: /usr/src/node-red/.local/share/virtualenvs/testrepo2-Ux6A5Ksm
Pipfile.lock not found, creating...
Locking [dev-packages] dependencies...
Building requirements...
Resolving dependencies...
✔ Success!
Locking [packages] dependencies...
Building requirements...
⠸ Locking...fatal: not a valid object name: HEAD
ERROR:setuptools_scm.file_finder_git:listing git files failed - pretending there aren't any
Traceback (most recent call last):
  File "/usr/lib/python3.8/tarfile.py", line 2311, in next
    tarinfo = self.tarinfo.fromtarfile(self)
  File "/usr/lib/python3.8/tarfile.py", line 1103, in fromtarfile
    obj = cls.frombuf(buf, tarfile.encoding, tarfile.errors)
  File "/usr/lib/python3.8/tarfile.py", line 1039, in frombuf
    raise EmptyHeaderError("empty header")
tarfile.EmptyHeaderError: empty header

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/tmp/testrepo2/.eggs/setuptools_scm-5.0.1-py3.8.egg/setuptools_scm/file_finder_git.py", line 51, in _git_ls_files_and_dirs
    return _git_interpret_archive(proc.stdout, toplevel)
  File "/tmp/testrepo2/.eggs/setuptools_scm-5.0.1-py3.8.egg/setuptools_scm/file_finder_git.py", line 32, in _git_interpret_archive
    with tarfile.open(fileobj=fd, mode="r|*") as tf:
  File "/usr/lib/python3.8/tarfile.py", line 1629, in open
    t = cls(name, filemode, stream, **kwargs)
  File "/usr/lib/python3.8/tarfile.py", line 1510, in __init__
    self.firstmember = self.next()
  File "/usr/lib/python3.8/tarfile.py", line 2326, in next
    raise ReadError("empty file")
tarfile.ReadError: empty file
Resolving dependencies...
✔ Success!
Updated Pipfile.lock (b78cf0)!
```

Checking the .git directory, seems to be fine, not sure what is the issue here?

```
(testrepo2) /tmp/testrepo2 $ ls -larth
total 180K
drwxr-xr-x    2 node-red node-red    4.0K Mar  1 19:15 tests
-rw-r--r--    1 node-red node-red    2.1K Mar  1 19:15 setup.py
-rw-r--r--    1 node-red node-red     639 Mar  1 19:15 setup.cfg
drwxr-xr-x    4 node-red node-red    4.0K Mar  1 19:15 docs
-rw-r--r--    1 node-red node-red    7.6K Mar  1 19:15 README.rst
-rw-r--r--    1 node-red node-red    1.1K Mar  1 19:15 LICENSE
-rw-r--r--    1 node-red node-red     134 Mar  1 19:15 CHANGELOG.rst
-rw-r--r--    1 node-red node-red    1.2K Mar  1 19:15 .gitignore
-rw-r--r--    1 node-red node-red     291 Mar  1 19:15 .editorconfig
-rw-r--r--    1 node-red node-red     277 Mar  1 19:16 Pipfile
-rw-r--r--    1 node-red node-red     538 Mar  1 19:17 tox.ini
-rw-r--r--    1 node-red node-red     100 Mar  1 19:18 pyproject.toml
drwxr-xr-x    3 node-red node-red    4.0K Mar  1 19:18 .eggs
-rw-r--r--    1 node-red node-red   32.7K Mar  1 19:18 Pipfile.lock
drwxr-xr-x    4 node-red node-red    4.0K Mar  1 19:20 src
drwxr-xr-x    7 node-red node-red    4.0K Mar  1 19:20 .git
drwxrwxrwt    1 root     root       20.0K Mar  1 19:21 ..
drwxr-xr-x    3 node-red node-red    4.0K Mar  1 19:21 .pytest_cache
-rw-r--r--    1 node-red node-red   52.0K Mar  1 19:21 .coverage
drwxr-xr-x    8 node-red node-red    4.0K Mar  1 19:21 .
(testrepo2) /tmp/testrepo2 $ cd .git
(testrepo2) /tmp/testrepo2/.git $ ls -larth
total 40K
drwxr-xr-x    4 node-red node-red    4.0K Mar  1 19:15 refs
drwxr-xr-x    4 node-red node-red    4.0K Mar  1 19:15 objects
drwxr-xr-x    2 node-red node-red    4.0K Mar  1 19:15 info
drwxr-xr-x    2 node-red node-red    4.0K Mar  1 19:15 hooks
-rw-r--r--    1 node-red node-red      73 Mar  1 19:15 description
-rw-r--r--    1 node-red node-red      92 Mar  1 19:15 config
drwxr-xr-x    2 node-red node-red    4.0K Mar  1 19:15 branches
-rw-r--r--    1 node-red node-red      23 Mar  1 19:15 HEAD
drwxr-xr-x    7 node-red node-red    4.0K Mar  1 19:20 .
drwxr-xr-x    8 node-red node-red    4.0K Mar  1 19:21 ..
(testrepo2) /tmp/testrepo2/.git $ cat HEAD
ref: refs/heads/master
(testrepo2) /tmp/testrepo2/.git $
```

**Comment by sedelmeyer** [comment]
I was able to reproduce your errors above (using py38 instead of py37, just as you had specified).

I found that, after running `git init`, if I made an initial commit (such as `git add -A` and `git commit -m "first commit"`) the following error no longer occurs while runing `pipenv install --dev`:
```
ERROR:setuptools_scm.file_finder_git:listing git files failed - pretending there aren't any
```
I must have overlooked that in the past. It turns out that `setuptools_scm` is a lot happier if your git repository has at least one commit to it prior to installing the local package using `pipenv`.

Let me know if that fix solves the issue for you.

I'll be certain to make an update to my documentation to account for this added step during set up.
